src/member/guards/group-manager.guard.ts

```diff
@@ -16,11 +16,9 @@ export class GroupManagerGuard implements CanActivate {
   async canActivate(context: ExecutionContext): Promise<boolean> {
     const request = context.switchToHttp().getRequest<Request>();
     const groupId: number = Number(request.params.groupId);
-    const userId: number | undefined = (request.user as GroupMember)?.userId;
+    const user = request.user as AuthenticatedUser;
+    const userId: number = user.userId;
```

Comment:
```suggestion
    const userId = user.userId;
```
const로 선언되는 원시타입(number, string, boolean 같은 타입)은 변수에 할당되는 값으로 인해 자동으로 타입이 바인딩되기 떄문에 보통 타입 명시를 안해줍니다

Reply:
제가 놓치고 있었던 거네요 const 변수는 선언할 때 이미 값이 정해져 있어서 그 값에 맞춰 타입이 자동으로 정해진다는 점을 다시 떠올리게 됐습니다 감사합니다!

Reply:
typescript-eslint를 사용하면 에디터에서 에러로 보여줘서 적용을 고려할 수 있겠네요
https://typescript-eslint.io/